dropwizard-metrics4/src/main/java/feign/metrics4/BaseMeteredClient.java

```diff
@@ -65,6 +66,8 @@ protected void recordFailure(RequestTemplate template, FeignException e) {
                 httpResponseCode(template),
                 "exception_name",
                 e.getClass().getSimpleName(),
+                "root_cause_name",
+                ExceptionUtils.getRootCause(e).getClass().getSimpleName(),
```

Comment:
`getCause` can return null, in that case would cause NPE

Reply:
As long as e is not null, getRootCause won’t return null as we can see here
https://github.com/OpenFeign/feign/pull/1883/files#diff-3bfdf878b3ea38d5268a9d72389c1f54ebabf1d76ad9a308e1eed93d95ed0048R38

Reply:
Ow, awesome, great!